### PR TITLE
fix settings.json schema to allow empty stings to

### PR DESCRIPTION
**Pull request by darkdarkdragon**
'emailHash' and 'walletLocator' fields so they
can be cleared

cover api/transaction/settings.js with tests

### CODE REVIEW

File: src/api/common/schemas/settings.json [hunk 1]
```diff
@@ -12,8 +12,18 @@
     "noFreeze": {"type": "boolean"},
     "globalFreeze": {"type": "boolean"},
     "defaultRipple": {"type": "boolean"},
-    "emailHash": {"$ref": "hash128"},
-    "walletLocator": {"$ref": "hash256"},
+    "emailHash": {
+      "oneOf": [
+        {"type": "string", "maxLength": 0},
```
Comment (clark800) [inline src/api/common/schemas/settings.json]: Do you think it would be better if we made `null` the value to clear the field?

Comment (darkdarkdragon) [inline src/api/common/schemas/settings.json]: Yes, that would be good - right now you can't clear integer fields - but I don't know how schema validator will react.

File: src/api/common/schemas/settings.json [hunk 2]
```diff
@@ -12,12 +12,22 @@
...
+    },
+    "walletLocator": {
+      "oneOf": [
+        {"type": "null"},
+        {"$ref": "hash256"}
+      ]
+    },
+    "walletSize": {"type": ["integer", "null"]},
     "messageKey": {"type": "string"},
     "domain": {"type": "string"},
```
Comment (clark800) [inline src/api/common/schemas/settings.json]: Do you know if these other non-boolean fields are clearable?

Comment (darkdarkdragon) [inline src/api/common/schemas/settings.json]: I've just took information from [here](https://github.com/ripple/ripple-lib/blob/develop/src/api/common/constants.js#L29) - one with 'defaults' field are clearable. Other from that - don't know.

**Comment by clark800** [comment]
LGTM

**Pull request merged**